src/overlays/actors/ovl_En_Famos/z_en_famos.c

```diff
@@ -28,133 +62,811 @@ const ActorInit En_Famos_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808ADFF0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808AE030.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/EnFamos_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808AE304.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808AE3A8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/func_808AE3FC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Famos/EnFamos_Draw.s")
+static s32 sAnimatedMaterialsDesgmented = false;
+
+void EnFamos_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnFamos* this = THIS;
+    Path* path;
+    s32 i;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    if (GET_FAMOS_PATH(thisx) != 0xFF) {
+        path = &globalCtx->setupPathList[this->actor.params];
+        this->pathPoints = Lib_SegmentedToVirtual(path->points);
+        this->pathNodeCount = path->count;
+        if (this->pathNodeCount == 1) {
+            this->pathPoints = NULL;
+            this->pathNodeCount = 0;
+        }
+    }
+
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawSquare, 30.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &gFamosSkeleton, &gFamosIdleAnim, this->jointTable, this->morphTable,
+                   FAMOS_LIMB_MAX);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider1, &this->actor, &sCylinderInit1);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider2, &this->actor, &sCylinderInit2);
+    Collider_InitAndSetJntSph(globalCtx, &this->emblemCollider, &this->actor, &sJntSphInit,
+                              this->emblemColliderElements);
+
+    if (!sAnimatedMaterialsDesgmented) { // init animated materials
+        for (i = 0; i < ARRAY_COUNT(sEmblemAnimatedMats); i++) {
+            sEmblemAnimatedMats[i] = Lib_SegmentedToVirtual(sEmblemAnimatedMats[i]);
+        }
+        sAnimatedMaterialsDesgmented = true;
+    }
+
+    this->actor.colChkInfo.mass = 250;
+    this->baseHeight = this->actor.world.pos.y;
+    // params: [this->actor.shape.rot.x] is used to set agro distance
+    this->aggroDistance = (this->actor.shape.rot.x <= 0) ? (200.0f) : (this->actor.shape.rot.x * 40.0f * 0.1f);
+    this->actor.shape.rot.x = 0;
+    this->actor.world.rot.x = 0;
+    this->hasFinishedRotating = true;
+    this->isCalm = true;
+    if (this->pathPoints != NULL) {
+        EnFamos_SetupPathingIdle(this);
+    } else {
+        EnFamos_SetupStillIdle(this);
+    }
+}
+
+void EnFamos_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnFamos* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider1);
+    Collider_DestroyCylinder(globalCtx, &this->collider2);
+    Collider_DestroyJntSph(globalCtx, &this->emblemCollider);
+}
+
+/**
+ * Summary: Sets 20 rocks to draw from an explosion on the ground (slam attack).
```

Comment:
You don't have to write "Summary" every time: if it's not detailed documentation you can just use `@brief` or make the comment.